scaleway/data_source_lb.go

```diff
@@ -0,0 +1,67 @@
+package scaleway
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/scaleway/scaleway-sdk-go/api/lb/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func dataSourceScalewayLb() *schema.Resource {
+	// Generate datasource schema from resource
+	dsSchema := datasourceSchemaFromResourceSchema(resourceScalewayLb().Schema)
+
+	// Set 'Optional' schema elements
+	addOptionalFieldsToSchema(dsSchema, "name", "region")
+
+	dsSchema["name"].ConflictsWith = []string{"lb_id"}
+	dsSchema["lb_id"] = &schema.Schema{
+		Type:          schema.TypeString,
+		Optional:      true,
+		Description:   "The ID of the load-balancer",
+		ValidateFunc:  validationUUIDorUUIDWithLocality(),
+		ConflictsWith: []string{"name"},
+	}
+
+	return &schema.Resource{
+		ReadContext: dataSourceScalewayLbRead,
+		Schema:      dsSchema,
+	}
+}
+
+func dataSourceScalewayLbRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api, region, err := lbAPIWithRegion(d, meta)
```

Comment:
lb do not support region anymore since version 2.1.0. I'm gonna update this branch.